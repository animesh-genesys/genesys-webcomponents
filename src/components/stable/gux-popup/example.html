<h1>gux-popup</h1>
<gux-popup>
  <gux-button
    slot="target"
    onClick='(function(event) {
      const popup = event.target.closest("gux-popup");
      popup.expanded = !popup.expanded;
    })(event)'
    >Target</gux-button
  >
  <gux-list slot="popup">
    <gux-list-item>Item 1</gux-list-item>
    <gux-list-item>Item 2</gux-list-item>
    <gux-list-item>Item 3</gux-list-item>
    <gux-list-item>Item 4</gux-list-item>
    <gux-list-item>Item 5</gux-list-item>
    <gux-list-item>Item 6</gux-list-item>
  </gux-list>
</gux-popup>
