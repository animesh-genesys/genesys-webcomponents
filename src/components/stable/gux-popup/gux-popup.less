@import (reference) '../../../style/color.less';

:host {
  isolation: isolate;
}

.gux-target-container {
  &.gux-disabled {
    pointer-events: none;
    cursor: default;
    opacity: 0.5;
  }
}

.gux-popup-container {
  z-index: 1;
  visibility: hidden;

  &.gux-expanded {
    visibility: visible;
  }
}
