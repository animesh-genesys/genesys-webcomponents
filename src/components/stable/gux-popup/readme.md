# gux-input



<!-- Auto Generated Below -->


## Properties

| Property   | Attribute  | Description | Type      | Default |
| ---------- | ---------- | ----------- | --------- | ------- |
| `disabled` | `disabled` |             | `boolean` | `false` |
| `expanded` | `expanded` |             | `boolean` | `false` |


## Slots

| Slot       | Description              |
| ---------- | ------------------------ |
| `"popup"`  | Required slot for popup  |
| `"target"` | Required slot for target |


## Dependencies

### Used by

 - [gux-action-button](../gux-action-button)
 - [gux-button-multi](../../beta/gux-button-multi)
 - [gux-dropdown-v2-beta](../../beta/gux-dropdown-v2)

### Graph
```mermaid
graph TD;
  gux-action-button --> gux-popup
  gux-button-multi --> gux-popup
  gux-dropdown-v2-beta --> gux-popup
  style gux-popup fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
