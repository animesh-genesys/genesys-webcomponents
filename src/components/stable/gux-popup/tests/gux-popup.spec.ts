import { newSpecPage } from '@stencil/core/testing';
import { GuxPopup } from '../gux-popup';

const components = [GuxPopup];
const language = 'en';

function getGuxPopupHtml(expanded: boolean, disabled: boolean): string {
  return `
  <gux-popup ${expanded ? 'expanded' : ''} ${disabled ? 'disabled' : ''}>
    <gux-button slot="target">Target</gux-button>
    <gux-list slot="popup">
      <gux-list-item>Item 1</gux-list-item>
      <gux-list-item>Item 2</gux-list-item>
      <gux-list-item>Item 3</gux-list-item>
    </gux-list>
  </gux-popup>
`;
}

describe('gux-popup', () => {
  describe('#render', () => {
    [
      { expanded: false, disabled: false },
      { expanded: true, disabled: false },
      { expanded: false, disabled: true },
      { expanded: true, disabled: true }
    ].forEach(({ expanded, disabled }, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const html = getGuxPopupHtml(expanded, disabled);
        const page = await newSpecPage({ components, html, language });

        expect(page.root).toMatchSnapshot();
      });
    });
  });
});
