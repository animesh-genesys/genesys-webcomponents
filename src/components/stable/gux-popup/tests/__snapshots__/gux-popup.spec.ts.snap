// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-popup #render should render component as expected (1) 1`] = `
<gux-popup>
  <mock:shadow-root>
    <div class="gux-target-container">
      <slot name="target"></slot>
      <div class="gux-popup-container" style="width: 0px; position: absolute; left: 0; top: 0; margin: 0;">
        <slot name="popup"></slot>
      </div>
    </div>
  </mock:shadow-root>
  <gux-button slot="target">
    Target
  </gux-button>
  <gux-list slot="popup">
    <gux-list-item>
      Item 1
    </gux-list-item>
    <gux-list-item>
      Item 2
    </gux-list-item>
    <gux-list-item>
      Item 3
    </gux-list-item>
  </gux-list>
</gux-popup>
`;

exports[`gux-popup #render should render component as expected (2) 1`] = `
<gux-popup expanded="">
  <mock:shadow-root>
    <div class="gux-target-container">
      <slot name="target"></slot>
      <div class="gux-expanded gux-popup-container" style="width: 0px; position: absolute; left: 0; top: 0; margin: 0;">
        <slot name="popup"></slot>
      </div>
    </div>
  </mock:shadow-root>
  <gux-button slot="target">
    Target
  </gux-button>
  <gux-list slot="popup">
    <gux-list-item>
      Item 1
    </gux-list-item>
    <gux-list-item>
      Item 2
    </gux-list-item>
    <gux-list-item>
      Item 3
    </gux-list-item>
  </gux-list>
</gux-popup>
`;

exports[`gux-popup #render should render component as expected (3) 1`] = `
<gux-popup disabled="">
  <mock:shadow-root>
    <div class="gux-disabled gux-target-container">
      <slot name="target"></slot>
      <div class="gux-popup-container" style="width: 0px; position: absolute; left: 0; top: 0; margin: 0;">
        <slot name="popup"></slot>
      </div>
    </div>
  </mock:shadow-root>
  <gux-button slot="target">
    Target
  </gux-button>
  <gux-list slot="popup">
    <gux-list-item>
      Item 1
    </gux-list-item>
    <gux-list-item>
      Item 2
    </gux-list-item>
    <gux-list-item>
      Item 3
    </gux-list-item>
  </gux-list>
</gux-popup>
`;

exports[`gux-popup #render should render component as expected (4) 1`] = `
<gux-popup disabled="" expanded="">
  <mock:shadow-root>
    <div class="gux-disabled gux-target-container">
      <slot name="target"></slot>
      <div class="gux-popup-container" style="width: 0px; position: absolute; left: 0; top: 0; margin: 0;">
        <slot name="popup"></slot>
      </div>
    </div>
  </mock:shadow-root>
  <gux-button slot="target">
    Target
  </gux-button>
  <gux-list slot="popup">
    <gux-list-item>
      Item 1
    </gux-list-item>
    <gux-list-item>
      Item 2
    </gux-list-item>
    <gux-list-item>
      Item 3
    </gux-list-item>
  </gux-list>
</gux-popup>
`;
